pkg/operation/botanist/namespaces.go

```diff
@@ -79,8 +79,9 @@ func (b *Botanist) DeploySeedNamespace(ctx context.Context) error {
 
 		// Label namespace to pin all control-plane pods of a shoot cluster to one zone
 		// if the seed has workers across different availability zones.
-		if zonePinningRequired(b.Shoot.GetInfo(), b.Seed.GetInfo()) && !metav1.HasLabel(namespace.ObjectMeta, v1beta1constants.ShootZonePinning) {
-			metav1.SetMetaDataLabel(&namespace.ObjectMeta, v1beta1constants.ShootZonePinning, "")
+		delete(namespace.Labels, v1beta1constants.ShootControlPlaneEnforceZone)
+		if zonePinningRequired(b.Shoot.GetInfo(), b.Seed.GetInfo()) && !metav1.HasLabel(namespace.ObjectMeta, v1beta1constants.ShootControlPlaneEnforceZone) {
+			metav1.SetMetaDataLabel(&namespace.ObjectMeta, v1beta1constants.ShootControlPlaneEnforceZone, "")
 		}
```

Comment:
```suggestion
		if !zonePinningRequired(b.Shoot.GetInfo(), b.Seed.GetInfo()) {
			delete(namespace.Labels, v1beta1constants.ShootControlPlaneEnforceZone)
		} else if !metav1.HasLabel(namespace.ObjectMeta, v1beta1constants.ShootControlPlaneEnforceZone) {
			metav1.SetMetaDataLabel(&namespace.ObjectMeta, v1beta1constants.ShootControlPlaneEnforceZone, "")
		}
```

Otherwise, the label would be changed back and forth during reconciliation, if enforce zone is activated.
This function would set its value to `""`, but `AddZoneInformationToSeedNamespace` would add a zone value again later.

Reply:
Nice catch @oliver-goetz. I fixed it, PTAL.